### PR TITLE
update the manual page on records

### LINKED ISSUE #2100: Update language manual with new record types
The page on [records](https://apalache.informal.systems/docs/lang/records.html) needs an update

### PR DESCRIPTION
**Pull request by konnov**
Closes #2100. This is an update of the manual page on records. I have added a few examples to demonstrate unsafe access and row types.

### CODE REVIEW

File: test/tla/cli-integration-tests.md [hunk 1]
```diff
@@ -3255,6 +3255,30 @@ Typing input error: Missing @typeAlias: newAlias = <type>;
...
+[TestGetX.tla:2:12-2:14]: Cannot apply r to the argument "x" in r["x"].
+```
+
+### typecheck TestGetXWithRows.tla
+
+```sh
+$ $ apalache-mc typecheck TestGetXWithRows.tla | sed 's/[IEW]@.*//'
+...
+EXITCODE: OK
+```
```
Comment (shonfeder) [inline test/tla/cli-integration-tests.md:3280]:
```suggestion
```sh
$ apalache-mc typecheck TestUnsafeRecord.tla | sed 's/[IEW]@.*//'
...
[TestUnsafeRecord.tla:6:17-6:19]: Cannot apply R() to the argument "b" in R()["b"]
```

### typecheck TestGetX.tla

```sh
$ apalache-mc typecheck TestGetX.tla | sed 's/[IEW]@.*//'
...
[TestGetX.tla:2:12-2:14]: Cannot apply r to the argument "x" in r["x"].
```

### typecheck TestGetXWithRows.tla

```sh
$ apalache-mc typecheck TestGetXWithRows.tla | sed 's/[IEW]@.*//'
...
EXITCODE: OK
```
```

File: test/tla/cli-integration-tests.md [hunk 2]
```diff
@@ -3255,6 +3255,30 @@ Typing input error: Missing @typeAlias: newAlias = <type>;
 EXITCODE: ERROR (255)
 ```
 
+### typecheck TestUnsafeRecord.tla
+
+```sh
+$ $ apalache-mc typecheck TestUnsafeRecord.tla | sed 's/[IEW]@.*//'
```
Comment (shonfeder) [inline test/tla/cli-integration-tests.md:3261]:
```suggestion
$ apalache-mc typecheck TestUnsafeRecord.tla | sed 's/[IEW]@.*//'
```

File: test/tla/cli-integration-tests.md [hunk 3]
```diff
@@ -3255,6 +3255,30 @@ Typing input error: Missing @typeAlias: newAlias = <type>;
...
+...
+[TestUnsafeRecord.tla:6:17-6:19]: Cannot apply R() to the argument "b" in R()["b"]
+```
+
+### typecheck TestGetX.tla
+
+```sh
+$ $ apalache-mc typecheck TestGetX.tla | sed 's/[IEW]@.*//'
```
Comment (shonfeder) [inline test/tla/cli-integration-tests.md]:
```suggestion
$ apalache-mc typecheck TestGetX.tla | sed 's/[IEW]@.*//'
```

File: test/tla/cli-integration-tests.md [hunk 4]
```diff
@@ -3255,6 +3255,30 @@ Typing input error: Missing @typeAlias: newAlias = <type>;
...
+...
+[TestGetX.tla:2:12-2:14]: Cannot apply r to the argument "x" in r["x"].
+```
+
+### typecheck TestGetXWithRows.tla
+
+```sh
+$ $ apalache-mc typecheck TestGetXWithRows.tla | sed 's/[IEW]@.*//'
```
Comment (shonfeder) [inline test/tla/cli-integration-tests.md]:
```suggestion
$ apalache-mc typecheck TestGetXWithRows.tla | sed 's/[IEW]@.*//'
```

File: test/tla/TestGetXWithRows.tla [hunk 5]
```diff
@@ -0,0 +1,4 @@
+--------------------- MODULE TestGetXWithRows ---------------------------------
+\* @type: { x: a, b } => b;
+GetX(r) == r.x
```
Comment (shonfeder) [inline test/tla/TestGetXWithRows.tla:3]: hmmm, shouldn't the type for this be 

```suggestion
\* @type: { x: a, b } => a;
GetX(r) == r.x
```

?

**Review by shonfeder — COMMENTED** [review]
Just one question about the typing here ...

File: test/tla/TestGetXWithRows.tla (hunk 5 above)
Comment (konnov) [reply test/tla/TestGetXWithRows.tla:3]: Oh yes. It should be as you are saying. I was just testing a related issue and forgot to fix it back: https://github.com/informalsystems/apalache/issues/2102

**Review by shonfeder — APPROVED** [review]
LGTM!

File: test/tla/TestGetXWithRows.tla (hunk 5 above)
Comment (shonfeder) [reply test/tla/TestGetXWithRows.tla]: If this was type checking, does it indicate a bug? It's not clear to me how see how `a` and `b` could be unifiable here... Or is it right that a single field of a record can have a row type?

**Pull request merged**